tcms/bugs/tests/test_views.py

```diff
@@ -0,0 +1,89 @@
+# pylint: disable=too-many-ancestors
+from http import HTTPStatus
+from django.urls import reverse
+
+from tcms.tests.factories import BugFactory
+from tcms.tests import BaseCaseRun, user_should_have_perm
+from tcms.utils.permissions import initiate_user_with_default_setups
+
+
+class TestBugStatusChange(BaseCaseRun):
+    """Test the possible bug status changes.
+
+    Cases:
+    * Closing an open bug.
+    * Reopenning a closed bug.
+    * Try to reopen an already open bug.
+    * try to close an already closed bug.
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        initiate_user_with_default_setups(cls.tester)
+
+        cls.bug = '1'
+        cls.text = 'Test Comment'
+        cls.action = 'close'
+        cls.data = {
+            'bug': cls.bug,
```

Comment:
do you need this here at all? you are overriding it in each tests, so why define it here in the first place?

Reply:
Moved them under each test case.